src/main.cpp

```diff
@@ -3295,6 +3295,16 @@ bool CBlock::ConnectBlock(CTxDB& txdb, CBlockIndex* pindex, bool fJustCheck, boo
             // Only reject superblock when it is new And when QuorumHash of Block != the Popular Quorum Hash:
             if (IsLockTimeWithinMinutes(GetBlockTime(),15)  && !fColdBoot)
             {
+                CBitcoinAddress address(bb.GRCAddress);
+                bool validaddressinblock = address.IsValid();
+                if (!validaddressinblock)
+                {
+                    return error("ConnectBlock[] : Superblock staked with invalid GRC address in block");
+                }
+                if (!IsNeuralNodeParticipant(bb.GRCAddress, nTime))
+                {
+                    return error("ConnectBlock[] : Superblock staked by ineligible neural node participant");
```

Comment:
not fool proof that I agree on but anything open source has that as a greater disadvantage already. but its more steps it making it a little more work for the attacker to go through the hoops.